routes/baseDBMethods.js

```diff
@@ -0,0 +1,53 @@
+async function list(ctx, model) {
+    try {
+        var docs = await model.find({});
+        ctx.ok(docs);
+    } catch (error) {
+        ctx.internalServerError();
+    }
+}
+
+async function create(ctx, model) {
+    var document = new model(ctx.request.body);
+
+    try {
+        await document.save({
+            validateBeforeSave: false,
+        });
+        ctx.ok(document);
+    } catch (err) {
+        console.error(err);
+        if(err.name == 'ValidationError') {
+            ctx.badRequest('invalid document')    
+        } else {
+            ctx.internalServerError();
+        }
+    }
+}
+
+async function update(ctx, model) {
+    try {
+        var updatedDoc = await model.findOneAndUpdate({'_id': ctx.request.body._id }, 
+                                                      { $set: ctx.request.body }, 
+                                                      { new: true });
+        ctx.ok(updatedDoc);
+    } catch (error) {
```

Comment:
Can you check if you can tell the errors apart?

- For not-found ID, we should return (I think, but you better verify this) 404.
- For general server/db error, we should return 500.

This also applies for `remove()` method.